### ISSUE TITLE
FlutterErrorDetails context should be renamed to reason

### ISSUE THREAD

**Issue by feinstein** [flutter/flutter#64522]
`FlutterErrorDetails` has a parameter named `context`, this is misleading because `context` in Flutter usually means a `BuildContext`, whereas here's diagnostic data. 

I suggest this be renamed to `reason` as the `DiagnosticsNode` is supposed to be converted to English for an explanation of the error.

**Comment by mkw8263** [comment]
@pedromassangocode 
Please check the PR.

**Comment by feinstein** [comment]
@mkw8263 you need to wait for your PR to be accepted by one of the reviewers. This will take longer than usual as it's a breaking change. 

The reviewers are very busy so don't ask them to review it, unless more than a week has passed by and you didn't get any returns yet.

**Comment by goderbauer** [comment]
I don't think that making this breaking change is really worth it. It creates a lot of churn and the type of the argument clearly indicates that it is not a BuildContext.

**Comment by feinstein** [comment]
@goderbauer I confess I was not quite sure as well for the same reasons you described, but I decided to open it anyway to gather opinions. Also, it goes along with my Crashlytics PR [here](https://github.com/FirebaseExtended/flutterfire/pull/3334), which uses `FlutterErrorDetails` internally.

**Comment by jamesderlin** [comment]
I don't think that "context" should necessarily mean `BuildContext`.   It's a generic term that's used all the time (not just in Flutter).  The term "context" should be, ah, context-dependent. ;)

**Comment by feinstein** [comment]
In Android development `context` is always the application context or `Activity`. In the Flutter framework it's always the `BuildContext`.

I agree that "context" is a generic word, but still there are expectations and associations/assumptions that people make automatically, and we can avoid headaches by maintaining consistency.
